lib/open_api/processor/type.ex

```diff
@@ -8,6 +8,10 @@ defmodule OpenAPI.Processor.Type do
   @typedoc "Literal values, as found in `const` and `enum` definitions"
   @type literal :: binary | boolean | number | nil
 
+  @type boolean_format :: String.t()
+  @type integer_format :: String.t()
+  @type number_format :: String.t()
```

Comment:
I haven't decided whether to add all the formats mentioned in the OpenAPI specification:
- `int32` and `int64` for `integer` type
- `float` and `double` for `number` type